tests/specs/framework-detection.e2e.js

```diff
@@ -0,0 +1,61 @@
+import { supportsFetchExtended } from '../../tools/browser-matcher/common-matchers.mjs'
+import { getMetricsFromResponse } from './helpers/assertion-helpers'
+
+const config = {
+  init: {
+    page_view_timing: {
+      enabled: false
+    }
+  }
+}
+
+describe('framework detection', () => {
+  withBrowsersMatching(supportsFetchExtended)('detects a page built with REACT and sends a supportability metric', async () => {
+    const rumPromise = browser.testHandle.expectRum()
+    await browser.url(await browser.testHandle.assetURL('frameworks/react/simple-app/index.html', config)).then(() => browser.waitForAgentLoad())
+
+    await rumPromise.then(async () => {
+      browser.url(await browser.testHandle.assetURL('/'))
+      return browser.testHandle.expectMetrics(3000)
+    }).then((response) => {
+      const supportabilityMetrics = getMetricsFromResponse(response, true)
+      expect(supportabilityMetrics).toBeDefined()
+      expect(supportabilityMetrics.length).toBeGreaterThan(0) // SupportabilityMetrics objects were generated
+      const sm = supportabilityMetrics.find(x => x.params.name.includes('Framework'))
+      expect(sm.params.name).toEqual('Framework/React/Detected') // Supportability metric is React and is formatted correctly
+    })
```

Comment:
Could simplify this since we have a better expect library at our disposal.

```suggestion
    expect(supportabilityMetrics).toEqual(expect.arrayContaining([{
      params: { name: 'Framework/React/Detected' },
      stats: { t: expect.toBeFinite() }
    }]))
```